index.js

```diff
@@ -1,11 +1,277 @@
+        label: 'Basics',
+        fields: [
+          'title',
+          'slug',
+          'description',
+          'image',
+          'startDate',
+          'allDay',
+          'startTime',
+          'endTime'
+        ]
+      },
+      advanced: {
+        label: 'Advanced',
+        fields: ['dateType', 'endDate', 'repeatInterval', 'repeatCount']
+      },
+      meta: {
+        label: 'Meta',
+        fields: ['tags', 'published']
+      }
+    }
+  },
+  handlers(self, options) {
+    return {
+      beforeSave: {
+        async denormalizeDateTimes(req, piece, options) {
+          self.denormalizeDatesAndTimes(piece);
+        }
+      },
+      beforeInsert: {
+        setGroupId(req, piece, options) {
+          // Set groupId on parent if this is a repeating item
+          if (
+            piece.dateType === 'repeat' &&
+            !piece.groupId &&
+            !self._workflowPropagating
+          ) {
+            piece.groupId = self.apos.util.generateId();
+          }
+        }
+      },
+      afterInsert: {
+        async createRepeatItems(req, piece, options) {
+          if (self._workflowPropagating) {
+            // Workflow is replicating this but also its existing
+            // scheduled repetitions, don't re-replicate them and cause problems
+            return;
+          }
+          if (piece.dateType === 'repeat' && piece.aposMode === 'draft') {
+            await self.repeatEvent(req, piece, options);
+          }
+        }
+      },
+      afterPublish: {
+        async publishChildren(req, piece, options) {
+          // If this is a repeating item, publish its children also
+          if (piece.published.dateType === 'repeat' && piece.firstTime) {
+            const existing = await self.findChildren(req, {
+              groupId: piece.draft.groupId
+            });
+            for (const child of existing) {
+              if (!child.isClone) continue; // Skip the parent event
+              await self.publish(req, child, options);
+            }
+          }
+        }
+      }
+    };
+  },
+  methods(self, options) {
+    return {
+      denormalizeDatesAndTimes(piece) {
+        // Parse our dates and times
+        let startTime = piece.startTime;
+        const startDate = piece.startDate;
+        let endTime = piece.endTime;
+        let endDate;
+
+        if (piece.dateType === 'consecutive') {
+          endDate = piece.endDate;
+        } else {
+          piece.endDate = piece.startDate;
+          endDate = piece.startDate;
+        }
+
+        if (piece.allDay) {
+          startTime = '00:00:00';
+          endTime = '23:59:59';
+        }
+
+        if (piece.dateType === 'repeat') {
+          piece.hasClones = true;
+        }
+
+        piece.start = new Date(startDate + ' ' + startTime);
+        piece.end = new Date(endDate + ' ' + endTime);
+      },
+      async repeatEvent(req, piece, options) {
+        let i;
+        const repeat = parseInt(piece.repeatCount) + 1;
```

Comment:
I apologize if this is old 2.x logic, but please don't use the + 1 and then start the `for` loop at `1`. It would be more idiomatic javascript style to leave the integer as-is and start the loop at `0`.

Reply:
Removed the '+1' but I modified the loop to be `for (i = 1; i <= repeat; i++) {` as we need i to be at least 1 in this loop to create the first child event.